src/sentry/integrations/github/client.py

```diff
@@ -86,9 +87,7 @@ def get_tree(self, repo_full_name: str, tree_sha: str) -> JSONData:
                 logger.warning(
                     f"The tree for {repo_full_name} has been truncated. Use different a approach for retrieving contents of tree."
                 )
-            # tree -> list of mode, path, sha, type and url for file/directory
-            # XXX: We should optimize the data structure to be a tree from file to top src dir
-            tree = contents["tree"]
+            tree = trim_tree(contents["tree"], ["python"])
```

Comment:
This simply takes the tree object and reduces it to the list of Python files on a repo.